facia-purger/src/main/scala/com/gu/purge/facia/ParseS3Path.scala

```diff
@@ -0,0 +1,20 @@
+package com.gu.purge.facia
+
+import org.parboiled2.{ Parser, ParserInput, Rule1 }
+
+import scala.util.Success
+
+// /aws-frontend-store/CODE/frontsapi/pressed/live/au/fapi/pressed.json
+class ParseS3Path(stage: String, val input: ParserInput) extends Parser {
+  def prefix = rule { s"$stage/frontsapi/pressed/live/" }
+  val suffix = "/fapi/pressed.json"
+
+  // Capture the front by using a negative lookahead for the suffix
+  def front = rule { capture(oneOrMore(!suffix ~ ANY)) }
+  def expr = rule { prefix ~ front ~ suffix ~ EOI }
+
+  def apply() = expr.run() match {
+    case Success(matched) => Some(matched)
+    case x => { println(s"error: $x"); None }
```

Comment:
I was actually discussing about this with Alex.
Ideally yes, the question is how do we do it. There is 2 options:
A. Make the lambda send directly to Kinesis but it means adding dependencies into the lambda to handle this part
B. Use [Real-time Processing of Log Data with Subscriptions](http://docs.aws.amazon.com/AmazonCloudWatch/latest/DeveloperGuide/Subscriptions.html) aka automatic forward of Cloudwatch logs to Kinesis. 

Given that the cloudwatch logs comes for free with any Lambda, I tend to like option B. I wonder if it would be possible to Cloudform this real-time processing at the same time as the lambda to make it fully automated. I would need to look at the documentation a bit more